ui/apps/platform/src/Containers/NetworkGraph/NetworkGraph.tsx

```diff
@@ -28,76 +32,167 @@ const model: Model = {
             label: 'stackrox',
             style: { padding: 15 },
             data: {
                 collapsible: true,
                 showContextMenu: false,
+                entityType: 'NAMESPACE',
+            },
+        },
+        {
+            id: 'EXTERNAL',
+            type: 'group',
+            children: ['__MzQuMTIwLjAuMC8xNg', 'afa12424-bde3-4313-b810-bb463cbe8f90'],
+            group: true,
+            label: 'External to cluster',
+            style: { padding: 15 },
+            data: {
+                collapsible: true,
+                showContextMenu: false,
+                entityType: 'EXTERNAL',
             },
         },
     ],
     edges: [
         {
             id: 'e1',
             type: 'edge',
-            source: 'n1',
-            target: 'n2',
+            source: 'e337f873-64d8-46be-84ed-2a0c38c75fac',
+            target: '09134b5d-8c12-41e8-821b-c97a5a1331c9',
         },
         {
             id: 'e2',
             type: 'edge',
-            source: 'n2',
-            target: 'n1',
+            source: '09134b5d-8c12-41e8-821b-c97a5a1331c9',
+            target: 'e337f873-64d8-46be-84ed-2a0c38c75fac',
+        },
+        {
+            id: 'e3',
+            type: 'edge',
+            source: '__MzQuMTIwLjAuMC8xNg',
+            target: 'e337f873-64d8-46be-84ed-2a0c38c75fac',
+        },
+        {
+            id: 'e4',
+            type: 'edge',
+            source: 'afa12424-bde3-4313-b810-bb463cbe8f90',
+            target: 'e337f873-64d8-46be-84ed-2a0c38c75fac',
         },
     ],
 };
 
+function findEntityById(graphModel, id) {
+    let entity = null;
+    entity = graphModel.nodes?.find((node) => node.id === id);
+    if (!entity) {
+        entity = graphModel.groups?.find((group) => group.id === id);
+    }
+    return entity;
+}
+
+function getUrlParamsForEntity(selectEntity): [string, string] {
+    const urlDetailTypes = {
+        NAMESPACE: 'namespace',
+        DEPLOYMENT: 'deployment',
+        CIDR_BLOCK: 'cidr',
+        EXTERNAL_ENTITIES: 'internet',
+        EXTERNAL: 'external',
+    };
+    const detailType = urlDetailTypes[selectEntity.data.entityType];
+    const detailId = selectEntity.id;
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+    return [detailType, detailId];
+}
+
 const TopologyComponent = () => {
     const [selectedObj, setSelectedObj] = useState<NodeModel | EdgeModel | null>(null);
+    const history = useHistory();
 
     const controller = useVisualizationController();
 
     React.useEffect(() => {
         function onSelect(ids: string[]) {
             const newSelectedId = ids?.[0] || null;
             // check if selected id is for a node
-            const newSelectedNode = model.nodes?.find((node) => node.id === newSelectedId);
-            if (newSelectedNode) {
-                setSelectedObj(newSelectedNode);
+            // const newSelectedEntity = model.nodes?.find((node) => node.id === newSelectedId);
```

Comment:
Looks like we'll have to do this a bit differently because of this case:
1. Select a deployment
2. Refresh the page (the URL should include that deployment id)
3. It won't show the side panel or the deployment selected

I would expect that the side panel would be open to that deployment

Reply:
Also, maybe we shouldn't keep the `selectedObj` state because we'll have an issue with synchronizing data. We can infer the selected object using the URL state

Reply:
You're absolutely right--I should have noted in my description that is the end-goal, of which this is just a piece, so that we can work on related sections without getting merge conflicts.